business_rules/engine.py

```diff
@@ -15,6 +17,18 @@ def run_all(rule_list, defined_variables, defined_actions, stop_on_first_trigger
     return rule_was_triggered
 
 
+async def async_run_all(rule_list, defined_variables, defined_actions, stop_on_first_trigger=False):
+
+    rule_was_triggered = False
+    for rule in rule_list:
+        result = await async_run(rule, defined_variables, defined_actions)
+        if result:
+            rule_was_triggered = True
+            if stop_on_first_trigger:
+                return True
+    return rule_was_triggered
```

Comment:
```suggestion
async def async_run_all(rule_list, defined_variables, defined_actions, stop_on_first_trigger=False):
    if stop_on_first_trigger:
        for rule in rule_list:
            result = await async_run(rule, defined_variables, defined_actions)
            if result:
                return True
        return False
    else:
        _async_run = partial(async_run, defined_variables=defined_variables, defined_actions=defined_actions)
        results = asyncio.gather(*map(_async_run, rule_list))
        return any(results)
```

Than how about this?